### PR TITLE
KT-18482: "Move lambda argument to parenthesis" action generate uncompilable code fixed

### PR DESCRIPTION
**Pull request by Dimach**

### CODE REVIEW

File: idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt [hunk 1]
```diff
@@ -70,7 +72,11 @@ fun KtLambdaArgument.moveInsideParenthesesAndReplaceWith(
     val newCallExpression = oldCallExpression.copy() as KtCallExpression
 
     val psiFactory = KtPsiFactory(project)
-    val argument = if (newCallExpression.getValueArgumentsInParentheses().any { it.isNamed() }) {
+
+    val argument = if (newCallExpression.getValueArgumentsInParentheses().run {
```
Comment (yole) [inline idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt]: I think this nested `run` expression is far too complex. Better to extract this logic into a method.

File: idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt [hunk 2]
```diff
@@ -354,11 +360,11 @@ fun KtTypeParameterListOwner.addTypeParameter(typeParameter: KtTypeParameter): K
     val list = KtPsiFactory(this).createTypeParameterList("<X>")
     list.parameters[0].replace(typeParameter)
     val leftAnchor = when (this) {
-        is KtClass -> nameIdentifier ?: getClassOrInterfaceKeyword()
-        is KtNamedFunction -> funKeyword
-        is KtProperty -> valOrVarKeyword
-        else -> null
-    } ?: return null
+                         is KtClass -> nameIdentifier ?: getClassOrInterfaceKeyword()
```
Comment (piotrek1543) [inline idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt]: There is a lot of unnecessary code formatting, which make harder to find what exactly author of PR changed. @Dimach, did was intentional and done by you or automatically changed by IntelliJ?

File: idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt (hunk 1 above)
Comment (piotrek1543) [inline idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt]: I agree @yole. This part is really good example what should be extracted to method.

File: idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt (hunk 2 above)
Comment (Dimach) [inline idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt]: I use reformat code action (alt+shift+l) so idea format my changes and all file. I can remove all format changes from PR if you want.

Comment (yole) [inline idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt]: It's OK to keep the formatting changes.

Comment (piotrek1543) [inline idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt]: @Dimach I just wondered the reason of this changes. Now, it's ok.;-)

**Comment by Dimach** [comment]
Done.

File: idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt [hunk 3]
```diff
@@ -92,6 +95,14 @@ fun KtLambdaArgument.moveInsideParenthesesAndReplaceWith(
     return oldCallExpression.replace(newCallExpression) as KtCallExpression
 }
 
+private fun shouldLambdaParameterNamed(args: List<ValueArgument>, callExpr: KtCallExpression): Boolean {
+    return args.any { it.isNamed() } ||
```
Comment (yole) [inline idea/idea-core/src/org/jetbrains/kotlin/idea/core/psiModificationUtils.kt]: Now that it's a separate function, there's no need to cramp it all into a single expression.

```
if (args.any { it.isNamed () }) return true
val callee = ...resolve as? KtFunction ?: return true
return callee.valueParameters.any { ... }
```

Also the grammatically correct name would be `shouldLambdaParameterBeNamed`.

**Comment by yole** [comment]
Thanks a lot for your contribution!

**Pull request merged**